File: employee-management/employee-model/src/main/java/com/pulse/air/employee/contract/PositionService.java
package com.pulse.air.employee.contract;

import com.pulse.air.common.model.ApiException;
import com.pulse.air.common.model.ApiListResponse;
import com.pulse.air.common.model.ApiRequest;
import com.pulse.air.commons.contract.BaseCRUDService;
import com.pulse.air.employee.model.position.PositionRequest;
import com.pulse.air.employee.model.position.PositionResponse;

public interface PositionService extends BaseCRUDService<PositionResponse, PositionRequest> {

	public ApiListResponse<PositionResponse> findByEmployeeId(ApiRequest<Long> request) throws ApiException;
}

File: airpulse-ui/src/app/pages/admin/services/position.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable, map } from "rxjs";
import { devEnvironment } from "src/environments/devenv";
import { ApiListResponse } from "../../shared/model/api-response";
import { BaseService } from "../../shared/services/base.service";
import { LoaderService } from "../../shared/services/loader.service";
import { PositionResponse, PositionRequest } from "../model/position.model";


@Injectable({
    providedIn: 'root',
})
export class PositionService extends BaseService<PositionResponse, PositionRequest> {
    constructor(protected override http: HttpClient, protected override loader: LoaderService) {
        super(http, 'emp/position', loader);
    }

    getByEmployeeId(employeeId: number): Observable<PositionResponse[]> {
        this.loader.show();
        const url = `${devEnvironment.baseUrl}/emp/position/employee/${employeeId}`;
        return this.http.get<ApiListResponse<PositionResponse>>(url).pipe(
            map((response) => {
                this.loader.hide();
                return response['data'];
            })
        );
    }
}